core/application/impl/app_config_impl.cpp

```diff
@@ -0,0 +1,332 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "application/impl/app_config_impl.hpp"
+
+#include <rapidjson/document.h>
+#include <rapidjson/filereadstream.h>
+#include <boost/program_options.hpp>
+#include <iostream>
+
+namespace {
+  struct RegisteredHandlers {
+    using Handler =
+        void (kagome::application::AppConfigurationImpl::*)(rapidjson::Value &);
+    char const *segment_name;
+    Handler handler;
+  };
+
+  template <typename T, typename Func>
+  inline void find_argument(boost::program_options::variables_map &vm,
+                            char const *name,
+                            Func &&f) {
+    assert(nullptr != name);
+    auto it = vm.find(name);
+    if (it != vm.end()) std::forward<Func>(f)(it->second.as<T>());
```

Comment:
Please, use braces for any block, including single-statament block.

Reply:
https://google.github.io/styleguide/cppguide.html#Conditionals

Reply:
_In general, curly braces are not required for single-line statements, but **they are allowed if you like them**; conditional or loop statements with complex conditions or statements may be more readable with curly braces. Some projects require that an if must always have an accompanying brace._

Ok, agree. Tastes differ. =) Just, this seems more readable, for me.